Issue #19529: Fix unicode_aswidechar() with 4byte unicode and 2byte wchar_t, for AIX

--- Objects/unicodeobject.c
+++ Objects/unicodeobject.c
@@ -1267,7 +1267,7 @@
     Py_ssize_t nchar;
 
     u = PyUnicode_AS_UNICODE(unicode);
-    uend = u + PyUnicode_GET_SIZE(u);
+    uend = u + PyUnicode_GET_SIZE(unicode);
     if (w != NULL) {
         worig = w;
         wend = w + size;

